server/chat/background/task.py

```
@@ -455,6 +455,17 @@ def run_background_chat(
     logger.info(f"[BackgroundChat] Trigger: {trigger_metadata}")
 
+    # Skip re-delivered tasks (acks_late means Redis can re-queue before completion)
+    try:
+        redis_client = get_redis_client()
+        lock_key = f"bgchat:lock:{session_id}"
+        acquired = redis_client.set(lock_key, self.request.id, nx=True, ex=1800)
+        if not acquired:
+            logger.warning("[BackgroundChat] DUPLICATE DELIVERY: session %s already running — skipping.", session_id)
+            return {"session_id": session_id, "status": "skipped", "skipped": "duplicate_delivery"}
+    except Exception:
+        pass
```

Ruff (0.15.15)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)